src/main.js

```diff
@@ -112,7 +113,7 @@ function hideIfSponsored(e) {
   return possibleSponsoredTextQueries.some((query) => {
     const result = e.querySelectorAll(query);
     return [...result].some(t => {
-      const visibleText = getVisibleText(t).join("");
+      const visibleText = getVisibleText(t);
```

Comment:
I'm not sure I follow the logic here.

You change this because the sponsored text is now in `data-content`?

I think it could be cleaned up a bit more to make this more readable.

Reply:
I've changed `getVisibleText` to return always a string. Until now it could return either text (if the element didn't have any child or array if it was hidden or had children.

Reply:
I think `getVisibleText` returns an array of visible characters.

Reply:
Is there any reason you would prefer to get the array instead of the joined text (having in mind all the other discussions above)

Reply:
I don't have a preference. A string is fine too.